client/src/main/java/io/pravega/client/netty/impl/AppendBatchSizeTrackerImpl.java

```diff
@@ -52,25 +53,31 @@ public void recordAppend(long eventNumber, int size) {
         long last = lastAppendTime.getAndSet(now);
         lastAppendNumber.set(eventNumber);
         millisBetweenAppends.addNewSample(now - last);
+        appendsOutstanding.addNewSample(eventNumber - lastAckNumber.get());
         eventSize.addNewSample(size);
     }
 
     @Override
     public void recordAck(long eventNumber) {
         lastAckNumber.getAndSet(eventNumber);
+        appendsOutstanding.addNewSample(lastAppendNumber.get() - eventNumber);
     }
 
     /**
-     * Returns a block size that in an estimate of how much data will be written in the next {@link #TARGET_BATCH_TIME_MILLIS}
+     * Returns a block size that in an estimate of how much data will be written in the next
+     * {@link #MAX_BATCH_TIME_MILLIS} or half the server round trip time (whichever is less).
      */
     @Override
     public int getAppendBlockSize() {
         long numInflight = lastAppendNumber.get() - lastAckNumber.get();
         if (numInflight <= 1) {
             return 0;
         }
-        return (int) MathHelpers.minMax((long) (TARGET_BATCH_TIME_MILLIS / millisBetweenAppends.getCurrentValue()
-                * eventSize.getCurrentValue()), 0, MAX_BATCH_SIZE);
+        double appendsInMaxBatch = Math.max(1.0, MAX_BATCH_TIME_MILLIS / millisBetweenAppends.getCurrentValue());
+        double targetAppendsOutstanding = MathHelpers.minMax(appendsOutstanding.getCurrentValue() / 2.0, 1.0,
```

Comment:
Do these numbers signify anything? Any benefit into having them as constants? Does `2.0` here mean the same as the `2` in the `appendsOutstanding` declaration above?

Reply:
Those are unrelated. 
The "/ 2.0" to to get to the "half latency" discribed in the javadoc. I've changed this to "* 0.5" because that is a little more clear.
The 2 in 'appendsOutstanding' is in initial value that is used when we compute the rolling average of the number of events in flight. It is somewhat irrelevant as it will quickly change once we have some data. I chose 2 to be conservative, and 1 is too low as we already special case 1.

Reply:
ok.